### PR TITLE
Allow user to change page limit

### PR DESCRIPTION
**Pull request by PrashanthCorp**
Redoing [PR#66](https://github.com/Microsoft/vscode-cosmosdb/pull/66) - mixed up some commits there due to a rebase. 

package.json : expose max docs as global setting; 
nodes.ts: import that setting when using maxdocs

### CODE REVIEW

**Comment by PrashanthCorp** [comment]
@Eric : The reasoning was that when CosmosDB gets more API's on board, then this distinction would be required. I put the mongo as a qualifier if this setting is unique, and cosmosdb to mark the extension using it. 

Your thoughts?

**Comment by EricJizbaMSFT** [comment]
On one hand it is the CosmosDB extension, but on the other hand you can attach Mongo servers that aren't hosted in Azure and have nothing to do with CosmosDB. I would prefer to keep it consistent with the `mongo.shell.path` setting above that just prefixes it with mongo.

**Comment by EricJizbaMSFT** [comment]
But I also still think you could just do `cosmosDB.maxDocs`. The other apis should have similar behavior and we could use the same setting for each

File: package.json [hunk 1]
```diff
@@ -275,6 +275,11 @@
 					"type": "boolean",
 					"default": true,
 					"description": "Show or hide the Cosmos DB Explorer"
+				},
+				"cosmosDB.mongo.maxDocs" : {
+					"type":"integer",
+					"default":20,
```
Comment (whiskeyjay) [inline package.json]: space after colon

File: package.json [hunk 2]
```diff
@@ -275,6 +275,11 @@
 					"type": "boolean",
 					"default": true,
 					"description": "Show or hide the Cosmos DB Explorer"
+				},
+				"cosmosDB.mongo.maxDocs" : {
+					"type":"integer",
```
Comment (whiskeyjay) [inline package.json]: space after colon

File: package.json [hunk 3]
```diff
@@ -275,6 +275,11 @@
 					"type": "boolean",
 					"default": true,
 					"description": "Show or hide the Cosmos DB Explorer"
+				},
+				"cosmosDB.mongo.maxDocs" : {
```
Comment (whiskeyjay) [inline package.json]: no space before colon

File: src/mongo/nodes.ts [hunk 4]
```diff
@@ -268,7 +268,10 @@ export class MongoCollectionNode implements INode {
 	}
 
 	private find(args?: any): Thenable<string> {
-		const maxDocs = 20;
+		let maxDocs = <number>vscode.workspace.getConfiguration().get('cosmosDB.mongo.maxDocs');
```
Comment (whiskeyjay) [inline src/mongo/nodes.ts]: Is getConfiguration method guaranteed not to return null? Otherwise this might be a potential null ref error.

Comment (whiskeyjay) [inline src/mongo/nodes.ts]: The get method accepts a type parameter:

```
get<number>('cosmosDB.mongo.maxDocs');
```

File: src/mongo/nodes.ts [hunk 5]
```diff
@@ -268,7 +268,10 @@ export class MongoCollectionNode implements INode {
...
+		if(maxDocs <= 0){
```
Comment (whiskeyjay) [inline src/mongo/nodes.ts]: missing spaces:
```
if (maxDocs <= 0) {
```

Comment (whiskeyjay) [inline src/mongo/nodes.ts]: Could be one line:
```
maxDocs = maxDocs > 0 ? maxDocs : 20;
```

File: src/mongo/nodes.ts [hunk 6]
```diff
@@ -268,7 +268,15 @@ export class MongoCollectionNode implements INode {
 	}
 
 	private find(args?: any): Thenable<string> {
-		const maxDocs = 20;
+		let maxDocs: number = 0; 
+		try{
+		maxDocs = vscode.workspace.getConfiguration().get<number>('cosmosDB.mongo.maxDocs');
```
Comment (whiskeyjay) [inline src/mongo/nodes.ts]: indent

File: src/mongo/nodes.ts [hunk 7]
```diff
@@ -268,7 +268,15 @@ export class MongoCollectionNode implements INode {
...
+		}
+		catch(error){
+		}
+		finally{
+		maxDocs = maxDocs > 0 ? maxDocs : 20;
```
Comment (whiskeyjay) [inline src/mongo/nodes.ts]: indent

**Pull request merged**